app/user.go

```diff
@@ -0,0 +1,597 @@
+		RequestId:       ctx.String(RequestIDFlagName),
+	}
+	if req.ResourceVersion == "" {
+		req.ResourceVersion = user.ResourceVersion
+	}
+	resp, err := c.client.UpdateUser(c.ctx, req)
+	if err != nil {
+		return fmt.Errorf("unable to update user: %w", err)
+	}
+	return PrintProto(resp.GetRequestStatus())
+}
+
+func (c *UserClient) setAccountRole(
+	ctx *cli.Context,
+	userID string,
+	userEmail string,
+	accountRole string,
+) error {
+	user, userRoles, err := c.getUserAndRoles(userID, userEmail)
+	if err != nil {
+		return err
+	}
+	accountRoleToSet, err := getAccountRole(c.ctx, c.client, accountRole)
+	if err != nil {
+		return err
+	}
+	for i := range userRoles {
+		if userRoles[i].Id == accountRoleToSet.Id {
+			return fmt.Errorf("user already has '%s' account role", accountRoleToSet.Id)
+		}
+	}
+	// check if this is the global admin role, and replace all existing roles
+	if accountRoleToSet.Spec.AccountRole.ActionGroup == auth.ACCOUNT_ACTION_GROUP_ADMIN {
+		// set the user account admin role
+		y, err := ConfirmPrompt(ctx, "Setting admin role on user, please confirm")
+		if err != nil || !y {
+			return err
+		}
+		userRoles = []*auth.Role{accountRoleToSet}
+	} else {
+		for i := range userRoles {
+			ur := userRoles[i]
+			if ur.Type != auth.ROLE_TYPE_PREDEFINED {
+				continue
+			}
+			// find the current admin role to replace
+			ar := ur.Spec.AccountRole
+			if ar != nil && ar.ActionGroup != auth.ACCOUNT_ACTION_GROUP_UNSPECIFIED {
+				// only swap the current admin role
+				userRoles[i] = accountRoleToSet
+			}
+		}
+	}
+	roleNames := make([]string, len(userRoles))
+	for i := range userRoles {
+		roleNames[i] = userRoles[i].Id
+	}
+	user.Spec.Roles = roleNames
+	return c.performUpdate(ctx, user)
+}
+
+func (c *UserClient) updateUserNamespacePermissions(
+	ctx *cli.Context,
+	userID string,
+	userEmail string,
+	namespace string,
+	actionGroup auth.NamespaceActionGroup,
+) error {
+	user, err := c.getUser(userID, userEmail)
+	if err != nil {
+		return err
+	}
+	req := &authservice.UpdateUserNamespacePermissionsRequest{
+		Namespace: namespace,
+		UserNamespacePermissions: []*auth.UserNamespacePermissions{
+			{
+				UserId:      user.Id,
+				ActionGroup: actionGroup,
+			},
+		},
+		RequestId: ctx.String(RequestIDFlagName),
+	}
+	resp, err := c.client.UpdateUserNamespacePermissions(c.ctx, req)
+	if err != nil {
+		return fmt.Errorf("unable to update user's namespace permission: %w", err)
+	}
+	return PrintProto(resp.GetRequestStatus())
+}
+
+func (c *UserClient) setNamespacePermissions(
+	ctx *cli.Context,
+	userID string,
+	userEmail string,
+	namespacePermissions []string,
+) error {
+	npm, err := toNamespacePermissionsMap(namespacePermissions)
+	if err != nil {
+		return err
+	}
+	for namespace, permission := range npm {
+		ag, err := toNamespaceActionGroup(permission)
+		if err != nil {
+			return err
+		}
+		if err := c.updateUserNamespacePermissions(ctx, userID, userEmail, namespace, ag); err != nil {
```

Comment:
isn't this another case where we can do a batch update here?

Reply:
actually, I am pretty sure we HAVE to do this in a single batch

Reply:
nevermind, the above statement is wrong. That API can actually be used btw to update multiple permissions either for the same user or across users